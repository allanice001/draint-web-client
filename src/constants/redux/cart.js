export const GET_CART_ITEMS = 'GET_CART_ITEMS';

export const LOAD_CART_ITEMS = 'LOAD_CART_ITEMS';

export const SET_CART_ITEMS = 'SET_CART_ITEMS';

export const SET_CART_DATA = 'SET_CART_DATA';

export const SET_CART_TOTAL = 'SET_CART_TOTAL';

export const CALCULATING = 'CALCULATING';
