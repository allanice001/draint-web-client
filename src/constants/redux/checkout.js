export const START_CREATING_COLLECTOR_ACCOUNT = 'CREATE_COLLECTOR_ACCOUNT';

export const NEXT_STEP = 'NEXT_STEP';

export const STEP_BACK = 'STEP_BACK';

export const INITIAL_STEP = 'INITIAL_STEP';

export const CHANGE_PROCEED_VALUE = 'CHANGE_PROCEED_VALUE';

export const FETCHING = 'FETCHING';

export const FETCHED = 'FETCHED';

export const IS_CHANGED_INFO_FORM = 'IS_CHANGED_INFO_FORM';

export const IS_CHANGED_ADDRESS_FORM = 'IS_CHANGED_ADDRESS_FORM';

export const SET_INITIAL_STATE = 'SET_INITIAL_STATE';

export const SET_CHECKOUT_DATA = 'SET_CHECKOUT_DATA';

export const SET_ORDERS = 'SET_ORDERS';

export const SET_SHIPPING_COST = 'SET_SHIPPING_COST';

export const SET_STRIPE_TOKEN = 'SET_STRIPE_TOKEN';

export const SET_CART_ITEMS = 'SET_CART_ITEMS';

export const SET_SHIPMENT_RATE = 'SET_SHIPMENT_RATE';

export const GET_SHIPMENT_RATE = 'GET_SHIPMENT_RATE';

// Checkout payment system actions

export const SET_PAYMENT_SYSTEM_ACCESS = 'SET_PAYMENT_SYSTEM_ACCESS';

export const SET_PAYMENT_SYSTEM_IS_FROM_VALID =
  'SET_PAYMENT_SYSTEM_IS_FROM_VALID';
