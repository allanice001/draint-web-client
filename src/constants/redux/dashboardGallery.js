export const SET_GALLERY_LOADING = 'SET_GALLERY_LOADING';

export const GET_USER_ARTWORKS_SUCCESS = 'GET_USER_ARTWORKS_SUCCESS';

export const GET_USER_ARTWORKS_PAGINATION = 'GET_USER_ARTWORKS_PAGINATION';

export const GET_USER_AVATAR_SUCCESS = 'GET_USER_AVATAR_SUCCESS';

export const GET_SUBSCRIBED_ARTIST_SUCCESS = 'GET_SUBSCRIBED_ARTIST_SUCCESS';

export const SET_SUBSCRIBED_ARTIST_ALL = 'SET_SUBSCRIBED_ARTIST_ALL';

export const GET_SUBSCRIBED_ARTIST_PAGINATION = 'GET_SUBSCRIBED_ARTIST_PAGINATION';

export const SET_USER_AVATAR_LOADING = 'SET_USER_AVATAR_LOADING';

export const SET_BLOG_POSTS = 'SET_BLOG_POSTS';

export const SET_BLOG_POST = 'SET_BLOG_POST';

export const DELETE_SINGLE_BLOG_POST = 'DELETE_SINGLE_BLOG_POST';

export const OPEN_BLOG_EDITING_MODAL = 'OPEN_BLOG_EDITING_MODAL';

export const UPDATE_SINGLE_BLOG_POST = 'UPDATE_SINGLE_BLOG_POST';

export const SET_MODAL_TRUE = 'SET_MODAL_TRUE';

export const SET_MODAL_FALSE = 'SET_MODAL_FALSE';

export const SET_CONTACTS = 'SET_CONTACTS';

export const SET_CONTACT = 'SET_CONTACT';

export const UNSET_CONTACT = 'UNSET_CONTACT';

export const SET_EDIT_MODE_FALSE = 'SET_EDIT_MODE_FALSE';

export const SET_EDIT_MODE_TRUE = 'SET_EDIT_MODE_TRUE';

export const SET_RECIPIENTS = 'SET_RECIPIENTS';

export const UNSET_RECIPIENTS = 'UNSET_RECIPIENTS';

export const RESET_RECIPIENTS = 'RESET_RECIPIENTS';

export const SET_FOR_SALE_STATUS = 'SET_FOR_SALE_STATUS';

export const UPDATE_ARTWORK_PRICE = 'UPDATE_ARTWORK_PRICE';

export const SET_DELETE_MODE_TRUE = 'SET_DELETE_MODE_TRUE';

export const SET_DELETE_MODE_FALSE = 'SET_DELETE_MODE_FALSE';

export const SET_DELETE_CONTACT = 'SET_DELETE_CONTACT';

export const TOGGLE_HISTORY_LIST = 'TOGGLE_HISTORY_LIST';

export const TOGGLE_CONTACTS_LIST = 'TOGGLE_CONTACTS_LIST';

export const SET_ACTIVE_MESSAGE = 'SET_ACTIVE_MESSAGE';

export const SET_IS_CAN_USE = 'SET_IS_CAN_USE';

export const SET_MESSAGES = 'SET_MESSAGES';

export const SET_DELETE_MESSAGE = 'SET_DELETE_MESSAGE';

export const SET_PREVIEW_MODAL = 'SET_PREVIEW_MODAL';

export const GET_PREVIEW_TEMPLATE_SUCCESS = 'GET_PREVIEW_TEMPLATE_SUCCESS';

export const SET_DEFAULT_STATE = 'SET_DEFAULT_STATE';

export const SET_RESET_QUILL = 'SET_RESET_QUILL';

export const SET_ATTACHMENTS = 'SET_ATTACHMENTS';
