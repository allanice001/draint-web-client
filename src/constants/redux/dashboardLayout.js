export const SET_DASHBOARD_LOADING = 'SET_DASHBOARD_LOADING';

export const SET_USER_STATUS = 'SET_USER_STATUS';

export const ADD_BLOG_TAB = 'ADD_BLOG_TAB';
