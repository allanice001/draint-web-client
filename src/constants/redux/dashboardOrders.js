export const SET_ORDERS_LOADING = 'SET_ORDERS_LOADING';

export const GET_USER_ORDERS_SUCCESS = 'GET_USER_ORDERS_SUCCESS';

export const GET_USER_OFFERS_SUCCESS = 'GET_USER_OFFERS_SUCCESS';

export const PUT_ORDERS_INVOICE_SUCCESS = 'PUT_ORDERS_INVOICE_SUCCESS';

export const SET_ORDERS_LOADING_TRUE = 'SET_ORDERS_LOADING_TRUE';

export const SET_ORDERS_LOADING_FALSE = 'SET_ORDERS_LOADING_FALSE';

export const SET_AUCTION_MODAL = 'SET_AUCTION_MODAL';

export const SET_CURRENT_OFFER = 'SET_CURRENT_OFFER';

export const SET_OFFER_PAGINATION = 'SET_OFFER_PAGINATION';

// ========================================= //

export const FETCH_ORDERS = 'FETCH_ORDERS';

export const FETCH_OFFERS = 'FETCH_OFFERS';

export const FETCH_ORDERS_SUCCESS = 'FETCH_ORDERS_SUCCESS';

export const FETCH_OUT_COMING_ORDERS_SUCCESS = 'FETCH_OUT_COMING_ORDERS_SUCCESS';

export const FETCH_IN_COMING_ORDERS_SUCCESS = 'FETCH_IN_COMING_ORDERS_SUCCESS';

export const FETCH_OFFERS_SUCCESS = 'FETCH_OFFERS_SUCCESS';

export const FETCH_ORDERS_FAILED = 'FETCH_ORDERS_FAILED';

export const FETCH_OFFERS_FAILED = 'FETCH_OFFERS_FAILED';

export const GET_SHIPMENT_TRACK_SUCCESS = 'GET_SHIPMENT_TRACK_SUCCESS';

export const CLEAR_SHIPMENT_TRACK_SUCCESS = 'CLEAR_SHIPMENT_TRACK_SUCCESS';
