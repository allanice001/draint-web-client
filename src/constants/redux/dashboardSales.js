export const SET_SALES_DASHBOARD_LOADING = 'SET_SALES_DASHBOARD_LOADING';

export const SET_COLLECTORS_INCOMING_OFFERS_SUCCESS =
  'SET_COLLECTORS_INCOMING_OFFERS_SUCCESS';

export const SET_COLLECTORS_OUTCOMING_OFFERS_SUCCESS =
  'SET_COLLECTORS_OUTCOMING_OFFERS_SUCCESS';

export const FETCH_PAYOUTS_ORDERS = 'FETCH_PAYOUTS_ORDERS';

export const FETCH_PAYOUTS_ORDERS_SUCCESS = 'FETCH_PAYOUTS_ORDERS_SUCCESS';

export const SET_PAGE = 'SET_PAGE';

export const SET_PAGE_SIZE = 'SET_PAGE_SIZE';
