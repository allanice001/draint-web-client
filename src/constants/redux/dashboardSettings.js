export const SET_SETTINGS_LOADING = 'SET_SETTINGS_LOADING';

export const GET_USER_ACCOUNT_SUCCESS = 'GET_USER_ACCOUNT_SUCCESS';

export const GET_USER_SIGNATURE_SUCCESS = 'GET_USER_SIGNATURE_SUCCESS';

export const SET_IS_CREATED = 'SET_IS_CREATED';

export const SET_ARTIST_SETTINGS = 'SET_ARTIST_SETTINGS';
