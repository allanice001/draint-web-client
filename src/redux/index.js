import * as fromBilling from './billing/billing-reducer';
import * as fromBlog from './blog';

import { DELETE_USER_DATA_SUCCESS } from '../constants/redux/user';
import artistRootReducer from './artist/artistRootReduser';
import artistsRootReducer from './artists/artistsRootReducer';
import artworkRootReducer from './artwork/artworkRootReducer';
import artworksRootReducer from './artworks/artworksRootReducer';
import cartRootReducer from './cart/cart-root-reducer';
import checkoutReducer from './checkout/checkoutReducer';
import { combineReducers } from 'redux';
import dashboardRootReducer from './dashboard/dashboardRootReducer';
import faqReducer from './faq/reducers/faqReducer';
import filtersReducer from './global/filters/filtersReducer';
import { reducer as formReducer } from 'redux-form';
import homepageRootReducer from './homepage/homeRootReducer';
import legalReducer from './legal/reducers/legalReducer';
import masterRootReducer from './master/masterRootReducer';
import navbarReducer from './global/navbar/navbarReducer';
import notificationReducer from './global/notiifcation/notificationReducer';
import pricingReducer from './pricing/reducers/pricingReducer';
import searchReducer from './search';
import userReducer from './user';

const reducers = {
  user: userReducer,
  notification: notificationReducer,
  master: masterRootReducer,
  dashboard: dashboardRootReducer,
  artist: artistRootReducer,
  artists: artistsRootReducer,
  artwork: artworkRootReducer,
  artworks: artworksRootReducer,
  home: homepageRootReducer,
  filters: filtersReducer,
  navbar: navbarReducer,
  form: formReducer,
  checkout: checkoutReducer,
  pricing: pricingReducer,
  legal: legalReducer,
  cart: cartRootReducer,
  faq: faqReducer,
  blog: fromBlog.reducer,
  billing: fromBilling.reducer,
  search: searchReducer,
};

const appReducer = combineReducers(reducers);

const rootReducer = (state, action) => {
  if (action.type === DELETE_USER_DATA_SUCCESS) {
    state.form = undefined;
  }
  return appReducer(state, action);
};

export default rootReducer;
