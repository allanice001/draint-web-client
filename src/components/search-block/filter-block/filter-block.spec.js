import { applyMiddleware, createStore } from 'redux';

import FilterBlock from './filter-block';
import { Provider } from 'react-redux';
import React from 'react';
import renderer from 'react-test-renderer';
import thunk from 'redux-thunk';

describe('FilterBlock', () => {
  const initialState = {
    filters: {
      artworksSearch: {
        hashtag: [],
        style: [],
        medium: [],
        surface: [],
        country: [],
        completed: [],
      },
    },
  };

  const testProperties = {
    initialize: jest.fn(),
    filters: [],
    handleSubmit: jest.fn(),
    initialValues: {},
    results: 0,
    onClose: jest.fn(),
    isOpen: false,
    query: '',
  };

  const store = createStore(
    (state = initialState) => state,
    applyMiddleware(thunk)
  );

  const component = renderer.create(
    <Provider store={store}>
      <FilterBlock
        filters={testProperties.filters}
        handleSubmit={testProperties.handleSubmit}
        initialize={testProperties.initialize}
        initialValues={testProperties.initialValues}
        isOpen={testProperties.isOpen}
        onClose={testProperties.onClose}
        query={testProperties.query}
        results={testProperties.results}
      />
    </Provider>
  );

  it('renders correctly', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
