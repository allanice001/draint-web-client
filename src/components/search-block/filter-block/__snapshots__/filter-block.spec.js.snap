// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterBlock renders correctly 1`] = `
<div
  onTouchMove={[Function]}
  onTouchStart={[Function]}
>
  <form
    className="form"
    id="search-form"
    onSubmit={[MockFunction]}
  >
    <div
      className="header"
      id="search-header"
    >
      <span />
      <button
        onClick={[MockFunction]}
        type="button"
      >
        <svg
          className="close"
          fill="none"
          height={14}
          viewBox="0 0 22 22"
          width={14}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12.105 11l8.666-8.666a.781.781 0 00-1.105-1.105L11 9.895 2.334 1.23a.781.781 0 10-1.105 1.105L9.895 11 1.23 19.666a.781.781 0 101.105 1.105L11 12.105l8.666 8.666a.779.779 0 001.105 0 .781.781 0 000-1.105L12.105 11z"
            fill="#3F4041"
            stroke="#3F4041"
            strokeWidth="0.5"
          />
        </svg>
      </button>
    </div>
    <div
      className="filters"
    />
  </form>
  <div
    className="backdrop"
    onClick={[MockFunction]}
  />
</div>
`;
